### ISSUE TITLE
rust-bitcoin 0.29.2 does not build on MSRV

### ISSUE THREAD

**Issue by sanket1729** [rust-bitcoin/rust-bitcoin#1546]
I don't know what happened. Everything seems to be the same, but the feature resolver suddenly is working differently. Previously, it did transitively resolve secp dependencies correctly, but the build is failing now. 

For comparison, see two runs on rust-minsicript with the same code, the same rust and cargo version, but one is failing to compile. 
a) https://github.com/rust-bitcoin/rust-miniscript/actions/runs/3905748427/jobs/6673126310 (Fail)
b) https://github.com/rust-bitcoin/rust-miniscript/actions/runs/3897608468/jobs/6655458221 (Success)

The same dependencies are fetched in both cases. Just the feature resolver is working in different ways for 1.41.1. Everything works after 1.46, which surprisingly has no changes in the feature resolver. 

cc @apoelstra who spent a lot of time investigating this

**Comment by apoelstra** [comment]
@sanket1729 found https://github.com/rust-lang/cargo/pull/8395 which was merged between cargo 1.45 (which exhibits this problem) and cargo 1.46 (which doesn't). This seems to describe exactly our issue, though there's some indeterminism that we haven't zeroed in on.

But assuming this *is* the issue, the conclusion to draw is that this is simply a bug in cargo prior to 1.47, we got lucky that we didn't hit it before, and in the meantime we just need to patch around it and do a new point release of rust-secp with the patch.
